### ISSUE TITLE
url_open not working on Xcode 16 / iOS 18

### ISSUE THREAD

**Issue by nicolaigd** [YoYoGames/GameMaker-Bugs#8684]
### Description

The GameMaker function `url_open` doesn't work on iOS when building with Xcode 16 and running on iOS 18+. The game builds fine, but when running on a device nothing happens when calling the function.

In Xcode the following error is shown when using the function on a simulator:
`BUG IN CLIENT OF UIKIT: The caller of UIApplication.openURL(_:) needs to migrate to the non-deprecated UIApplication.open(_:options:completionHandler:). Force returning false (NO).`

So in short, openURL is deprecated and needs to be replaced.

Here is the sample package:
https://api.gamemaker.io/api/github/downloads/9bee0804-94f1-4c1e-a4ff-0a9555e1959d

### Steps To Reproduce

1. Call url_open on iOS 18+ built with Xcode 16
2. Nothing happens (error is shown in Xcode)

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.8.1 (Monthly)

### Which platform(s) are you seeing the problem on?

iOS

### Sample Package Attached?

- [x] I have attached my bug package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by jackerley** [comment]
Note for QA, this error only happens on simulator, on real device it opens the url just fine

**Comment by nicolaigd** [comment]
> Note for QA, this error only happens on simulator, on real device it opens the url just fine

No, the urls don't open on my physical iPhone 15 with iOS 18.1.1

This most definitely happens on real devices, it just depends on the versions of Xcode and iOS used, as I described.

**Comment by jackerley** [comment]
You will need to use a Beta release if you want a build with the fix in.  You can download from https://releases.gamemaker.io/release-notes/2024/1300

**Comment by harpwood** [comment]
> You will need to use a Beta release if you want a build with the fix in. You can download from https://releases.gamemaker.io/release-notes/2024/1300

Thank you for the tip. I initially thought the stable version would be live by the end of January, but it seems to have been postponed to March. So I might build with the beta release.

**Comment by clee2005** [comment]
I created a function in a custom extension which does the job for now.   It was easy to go through and change all the url_open() calls to use a script and if (os_type == os_ios) call extension else call url_open()

- (NSString *)Donkeysoft_OpenURL:(char *)urlString {
    // Convert char * to NSString
    NSString *urlNSString = [NSString stringWithUTF8String:urlString];
    if (!urlNSString) {
        return @"Invalid UTF-8 string.";
    }

    // Convert NSString to NSURL
    NSURL *url = [NSURL URLWithString:urlNSString];
    [[UIApplication sharedApplication] openURL:url options:@{} completionHandler:^(BOOL success) {
        if (success) {
            NSLog(@"URL opened successfully.");
        } else {
            NSLog(@"Failed to open URL.");
        }
    }];

    return @"Attempting to open URL.";
}

Tested and using this in our apps now.

**Comment by sihammill** [comment]
Fix appears to be fine in Beta IDE v2024.1300.0.769  Runtime v2024.1300.0.776
